### PR TITLE
[[Cleanup]] Header warning fixes from Emscripten project.

### PR DESCRIPTION
**Pull request by peter-b**

### CODE REVIEW

File: engine/src/stack.h [hunk 1]
```diff
@@ -1083,7 +1084,9 @@ class MCStack : public MCObject
 	void SetSystemWindow(MCExecContext& ctxt, bool setting);
 	void GetMetal(MCExecContext& ctxt, bool& r_setting);
 	void SetMetal(MCExecContext& ctxt, bool setting);
+	using MCObject::GetShadow;
```
Comment (runrevmark) [inline engine/src/stack.h:1087]: Why are these needed?

File: engine/src/stack3.cpp [hunk 2]
```diff
@@ -136,6 +136,12 @@ IO_stat MCStack::extendedload(MCObjectInputStream& p_stream, uint32_t p_version,
 	return t_stat;
 }
 
+IO_stat MCStack::load(IO_handle stream, uint32_t version)
+{
+	MCLog("%s should never be called!", __PRETTY_FUNCTION__);
+	MCUnreachable();
+}
+
 IO_stat MCStack::load(IO_handle stream, uint32_t version, uint1 type)
```
Comment (runrevmark) [inline engine/src/stack3.cpp:145]: I thought we had discussed renaming this method as it is not the same semantics as load(). (Refrains from ranting about why function name overloading is actually a bad thing for code readability and comprehension ;))

File: engine/src/stack.h (hunk 1 above)
Comment (peter-b) [inline engine/src/stack.h:1087]: Without the `using` declaration, the virtual method `MCObject::SetShadow(MCExecContext &, const MCInterfaceShadow &)` (fbase1) is masked by the overloaded virtual method `MCStack::SetShadow(MCExecContext &, bool)` (fderived2). In order to avoid ambiguity, either fbase1 needs to be used in the derived class (to create a vtable entry), or overridden with some implementation fderived1, or fderived2 needs to be renamed to remove the overload.  I went with option 1 because it makes the current behaviour explicit.

As a third option, I could rename `MCStack::SetShadow(MCExecContext &, bool)` to `MCStack::SetWindowShadow`.

File: engine/src/stack3.cpp (hunk 2 above)
Comment (peter-b) [inline engine/src/stack3.cpp:145]: Yes. However, it would still be a programming error to call `MCObject::load()` for an `MCStack` instance, so `load()` still needs to be overridden as above.  This is an interim fix.

File: engine/src/stack.h (hunk 1 above)
Comment (runrevmark) [inline engine/src/stack.h:1087]: Option 3 is preferable - it makes it completely obvious it is a different function without fiddling with scopes.

**Comment by livecode-vulcan** [comment]
📝 Hi @peter-b, I haven't been able to verify that you've signed our [LiveCode Contributor's Agreement](https://livecode.com/account/developer/contribute).

If you have previously signed the Contributor's Agreement, I may not be able to detect it because you haven't [linked your GitHub account to your LiveCode account](https://livecode.com/account/developer/github).

Please see the [information for contributors](https://github.com/runrev/livecode/blob/develop/HACKING.md) for more information.

**Comment by livecodefraser** [comment]
@livecode-vulcan review okay 0874bc8

I'm reviewing this as the discussion seems to have resolved to "this fix works for now" - if anyone has any objections, please let me know.

**Comment by livecode-vulcan** [comment]
💙 review by @livecodefraser ok 0874bc8517c0a65a6990af2e7e34db8a154ad894

**Comment by livecode-vulcan** [comment]
😞 test failure 0874bc8517c0a65a6990af2e7e34db8a154ad894
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/208)
- try-community-mac: [failure](https://vulcan.livecode.com/builders/try-community-mac/builds/207)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/208)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/203)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/210)
- try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/208)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/211)

**Comment by peter-b** [comment]
@livecodefraser Apparently it doesn't. ;-) I need to tweak this patch set anyway because some of the fixes were merged in separate pull requests.

**Comment by livecodesebastien** [comment]
@livecode-vulcan review not OK 0874bc8

**Comment by livecode-vulcan** [comment]
💔 review by @livecodesebastien not ok 0874bc8517c0a65a6990af2e7e34db8a154ad894

**Comment by peter-b** [comment]
These changes are now all addressed in other pull requests.
